doc/usage/extensions/autodoc.rst

```diff
@@ -299,6 +299,15 @@ inserting them into the page source under a suitable :rst:dir:`py:module`,
 
         .. autoclass:: module.name::Noodle
 
+   * For objects defined in multiple modules, autodoc will generate both
+     specific directives and use the ``:canonical:`` option accordingly.
+     This applies when importing a class from a submodule within a package's
+     __init__.py and adding it to the ``__all__`` attribute.
```

Comment:
I think adding to `__all__` is one of the use cases. It's not a requirement. I worry about this explanation confuses users.

Reply:
I'll try to reword this.  The idea was to give a example (which cover what seamed to be the most common use case).  This feature effectively adds one entry in the generated documentation, which is quite a change.